metricbeat/module/prometheus/collector/_meta/docs.asciidoc

```diff
@@ -1,5 +1,43 @@
-The Prometheus `collector` metricset fetches data from https://prometheus.io/docs/instrumenting/exporters/[prometheus exporters].
+The Prometheus `collector` metricset scrapes data from https://prometheus.io/docs/instrumenting/exporters/[prometheus exporters].
 
-All events with the same labels are grouped together as one event. The fields
-exported by this metricset vary depending on the Prometheus exporter that you're
-using.
+
+[float]
+=== Scraping from a Prometheus exporter
+
+To scrape metrics from a Prometheus exporter, configure the `hosts` field to it. The path
+to retrieve the metrics from (`/metrics` by default) can be configured with `metrics_path`.
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+- module: prometheus
+  period: 10s
+  hosts: ["localhost:9090"]
+  metrics_path: /metrics
+
+  #username: "user"
+  #password: "secret"
+
+  # This can be used for service account based authorization:
+  #bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+  #ssl.certificate_authorities:
+  #  - /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt
+-------------------------------------------------------------------------------------
+
+
+[float]
+=== Scraping all metrics from a Prometheus server
+
+This module can scrape all metrics stored in a Prometheus server, by using the
+https://prometheus.io/docs/prometheus/latest/federation/[federation API]. By pointing this
+config to the Prometheus server:
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+metricbeat.modules:
+- module: prometheus
```

Comment:
This is kind of like a second "flawor" of the metricset. In the future we potentially could add a simple config `federate.enabled: true` which will set the correct settings directly.

Reply:
I like it, it would be a nice addition, I agree it can be added later